bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java

```diff
@@ -331,6 +341,24 @@ public static void run(Path jbossHome, List<String> args, ModuleLoader moduleLoa
         // At this point we can configure JMX
         configureJMX(moduleClassLoader, bootableJar.log);
 
+        // Automatic loading of Security providers.
+        // Needed for logic that requires access to providers prior elytron subsystem is configured.
+        final ServiceLoader<Provider> providerServiceLoader = ServiceLoader.load(Provider.class, moduleClassLoader);
+        Iterator<Provider> iterator = providerServiceLoader.iterator();
+        SecurityManager sm = System.getSecurityManager();
+        for (;;) try {
+            if (! (iterator.hasNext())) break;
+            final Provider provider = iterator.next();
+            if (sm == null) {
+                new AddProviderAction(provider).run();
+            } else {
+                final Class<? extends Provider> providerClass = provider.getClass();
+                // each provider needs permission to install itself
+                doPrivileged(new AddProviderAction(provider), getProviderContext(providerClass));
+            }
```

Comment:
Is there a reason we do escalate to a privileged action here? There are other places in this entry point that would trigger SM failures if enabled. It seems odd to allow it in this specific case.

Reply:
I understand it this way:  Bootable JAR code base is granted all permissions. When adding a provider, the provider domain is used to restrict the set of permissions. FYI, this is the way it is implemented in WildFly boot sequence.